server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java

```diff
@@ -0,0 +1,330 @@
+            // register it once for watching.
+            enableSettingsWatcher(null, operatorSettingsDir().getParent());
+        } catch (Exception e) {
+            if (watchService != null) {
+                try {
+                    this.watchService.close();
+                } catch (Exception ignore) {} finally {
+                    this.watchService = null;
+                }
+            }
+
+            throw new IllegalStateException("unable to launch a new watch service", e);
+        }
+
+        this.watcherThreadLatch = new CountDownLatch(1);
+
+        new Thread(() -> {
+            try {
+                logger.info("file settings service up and running [tid={}]", Thread.currentThread().getId());
+
+                WatchKey key;
+                while ((key = watchService.take()) != null) {
+                    /**
+                     * Reading and interpreting watch service events can vary from platform to platform. E.g:
+                     * MacOS symlink delete and set (rm -rf operator && ln -s <path to>/file_settings/ operator):
+                     *     ENTRY_MODIFY:operator
+                     *     ENTRY_CREATE:settings.json
+                     *     ENTRY_MODIFY:settings.json
+                     * Linux in Docker symlink delete and set (rm -rf operator && ln -s <path to>/file_settings/ operator):
+                     *     ENTRY_CREATE:operator
+                     * Windows
+                     *     ENTRY_CREATE:operator
+                     *     ENTRY_MODIFY:operator
+                     * After we get an indication that something has changed, we check the timestamp, file id,
+                     * real path of our desired file.
+                     */
+                    if (Files.exists(settingsDir)) {
+                        try {
+                            Path path = operatorSettingsFile();
+
+                            if (logger.isDebugEnabled()) {
+                                key.pollEvents().stream().forEach(e -> logger.debug("{}:{}", e.kind().toString(), e.context().toString()));
+                            }
+
+                            key.pollEvents();
+                            key.reset();
+
+                            // We re-register the settings directory watch key, because we don't know
+                            // if the file name maps to the same native file system file id. Symlinks
+                            // are one potential cause of inconsistency here, since their handling by
+                            // the WatchService is platform dependent.
+                            settingsDirWatchKey = enableSettingsWatcher(settingsDirWatchKey, settingsDir);
+
+                            if (watchedFileChanged(path)) {
+                                processFileSettings(path, false);
+                            }
+                        } catch (Exception e) {
+                            logger.warn("error processing operator settings file", e);
```

Comment:
How is this possible given processFileSettings logs any errors? It seems like the only possible error here is IOException? Could this be narrowed to that exception, and the warning message be distinguished from that on line 307? They are very similarly worded, yet this one, I don't think is specific to processing the file, it's just some IO problem happened while trying read it.